silo-core/contracts/incentives/SiloIncentivesControllerGaugeLike.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {IERC20} from "openzeppelin5/token/ERC20/IERC20.sol";
+
+import {IGaugeLike} from "../interfaces/IGaugeLike.sol";
+import {SiloIncentivesController} from "./SiloIncentivesController.sol";
+import {ISiloIncentivesControllerGaugeLike} from "./interfaces/ISiloIncentivesControllerGaugeLike.sol";
+
+/// @dev Silo incentives controller that can be used as a gauge in the Gauge hook receiver
+contract SiloIncentivesControllerGaugeLike is SiloIncentivesController, ISiloIncentivesControllerGaugeLike {
+    /// @dev The share token of the gauge
+    address public immutable SHARE_TOKEN;
```

Comment:
why not `NOTIFIER`? isn't notifier a share token?

Reply:
The notifier should be a gaugeHookReceiver, as it will send notifications.

Reply:
in `DistributionManager` 
```
function _shareToken() internal view virtual returns (IERC20 shareToken) {
        shareToken = IERC20(NOTIFIER);
    }
```

so this is not only for sending notification, now we will have two methods for share tokens that returns different addresses?

Reply:
Why two? Still one method. We are overriding it in the SiloIncentivesControllerGaugeLike.

Reply:
resolved on call